mlir/lib/Catalyst/Transforms/DetectQNodes.cpp

```diff
@@ -161,36 +161,23 @@ LogicalResult DetectCallsInAsyncRegionsTransform::matchAndRewrite(LLVM::CallOp c
     return success();
 }
 
-/*
- *
- * Note:
- *     I will be using high level MLIR to describe the algorithm,
- *     but this transformation works at the LLVM dialect stage.
- *     Perhaps some of the passes could be made at higher level of abstractions.
- *
- * DetectCallsInAsyncRegionsTransform will match against:
- *
- *
- * into
- *
- * ```llvm
- * llvm.func @callee() attributes { qnode }
- *
- * llvm.func @foo() attributes { passthrough = [presplitcoroutine] } {
- *     llvm.call @callee() { catalyst.preInvoke }
- * }
- * ```
- *
- *
- */
-
+/* Next, we run DetectQnodeTransform */
 struct DetectQnodeTransform : public OpRewritePattern<LLVM::CallOp> {
     using OpRewritePattern<LLVM::CallOp>::OpRewritePattern;
 
     LogicalResult match(LLVM::CallOp op) const override;
     void rewrite(LLVM::CallOp op, PatternRewriter &rewriter) const override;
 };
 
+/* Here we only match with calls that have the { catalyst.preInvoke } annotations */
+LogicalResult DetectQnodeTransform::match(LLVM::CallOp callOp) const
+{
+    // The following is a valid match
+    //     llvm.call @callee() { catalyst.preInvoke }
+    bool validCandidate = isScheduledForTransformation(callOp);
+    return validCandidate ? success() : failure();
+}
+
```

Comment:
The "match" method changed its implementation. I would suggest moving this change to a different commit, because the commit description says it is related to "Documentation".

Reply:
Yes, due to the previous pass, which annotates:

```
        candidate &&
        hasQnodeAttribute(candidate.value()) 
 ```
 
 with `isScheduledForTransformation` the previous two checks are unnecessary.
 
 I can split the commit if you want.

Reply:
No need now. I just think separating the changes according to their nature would ease our review process.